endpoint_info.go

```diff
@@ -149,6 +151,33 @@ func (ep *endpoint) InterfaceNames() []driverapi.InterfaceNameInfo {
 	return iList
 }
 
+func (ep *endpoint) SetStaticRoutes(routes []types.StaticRoute) error {
+	ep.Lock()
+	defer ep.Unlock()
+
+	for _, r := range routes {
+		if r.RouteType == types.NEXTHOP {
+			ep.joinInfo.StaticRoutes = append(ep.joinInfo.StaticRoutes, &r)
+		} else {
+			if err := ep.AddInterfaceRoute(&r); err != nil {
```

Comment:
If this fails we need to revert the `StaticRoutes` added to `joinInfo` but if we are going to change the interface to `AddStaticRoute` then this comment may be moot.